algorithmic_efficiency/workloads/fastmri/fastmri_pytorch/models.py

```diff
@@ -24,38 +43,49 @@ class UNet(nn.Module):
   def __init__(self,
                in_chans: int = 1,
                out_chans: int = 1,
-               chans: int = 32,
+               num_channels: int = 32,
                num_pool_layers: int = 4,
-               dropout_rate: Optional[float] = 0.0) -> None:
+               dropout_rate: Optional[float] = 0.0,
+               use_tanh: bool = False,
+               use_layer_norm: bool = False) -> None:
     super().__init__()
 
     self.in_chans = in_chans
     self.out_chans = out_chans
-    self.chans = chans
+    self.num_channels = num_channels
     self.num_pool_layers = num_pool_layers
     if dropout_rate is None:
       dropout_rate = 0.0
-
-    self.down_sample_layers = nn.ModuleList(
-        [ConvBlock(in_chans, chans, dropout_rate)])
-    ch = chans
+    self.down_sample_layers = nn.ModuleList([
+        ConvBlock(in_chans,
+                  num_channels,
+                  dropout_rate,
+                  use_tanh,
+                  use_layer_norm)
+    ])
+    ch = num_channels
     for _ in range(num_pool_layers - 1):
-      self.down_sample_layers.append(ConvBlock(ch, ch * 2, dropout_rate))
+      self.down_sample_layers.append(
+          ConvBlock(ch, ch * 2, dropout_rate, use_tanh, use_layer_norm))
       ch *= 2
-    self.conv = ConvBlock(ch, ch * 2, dropout_rate)
+    self.conv = ConvBlock(ch, ch * 2, dropout_rate, use_tanh, use_layer_norm)
 
     self.up_conv = nn.ModuleList()
     self.up_transpose_conv = nn.ModuleList()
+
     for _ in range(num_pool_layers - 1):
-      self.up_transpose_conv.append(TransposeConvBlock(ch * 2, ch))
-      self.up_conv.append(ConvBlock(ch * 2, ch, dropout_rate))
+      self.up_transpose_conv.append(
+          TransposeConvBlock(ch * 2, ch, use_tanh, use_layer_norm))
+      self.up_conv.append(
+          ConvBlock(ch * 2, ch, dropout_rate, use_tanh, use_layer_norm))
       ch //= 2
 
-    self.up_transpose_conv.append(TransposeConvBlock(ch * 2, ch))
+    self.up_transpose_conv.append(
+        TransposeConvBlock(ch * 2, ch, use_tanh, use_layer_norm))
     self.up_conv.append(
         nn.Sequential(
-            ConvBlock(ch * 2, ch, dropout_rate),
-            nn.Conv2d(ch, self.out_chans, kernel_size=1, stride=1),
+            ConvBlock(ch * 2, ch, dropout_rate, use_tanh, use_layer_norm),
+            nn.Conv2d(ch, 1, kernel_size=1, stride=1),
```

Comment:
Is there a reason why `out_chans` is hard-coded now?

Reply:
Nope. Fixed.